snuba/admin/clickhouse/common.py

```diff
@@ -0,0 +1,62 @@
+from typing import MutableMapping
+
+from snuba import settings
+from snuba.clickhouse.native import ClickhousePool
+from snuba.clusters.cluster import ClickhouseClientSettings, ClickhouseCluster
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage
+from snuba.utils.serializable_exception import SerializableException
+
+
+class InvalidNodeError(SerializableException):
+    pass
+
+
+class InvalidCustomQuery(SerializableException):
+    pass
+
+
+class InvalidStorageError(SerializableException):
+    pass
+
+
+def is_valid_node(host: str, port: int, cluster: ClickhouseCluster) -> bool:
+    connection_id = cluster.get_connection_id()
+    return host == connection_id.hostname and port == connection_id.tcp_port
+
+
+CONNECTIONS: MutableMapping[StorageKey, ClickhousePool] = {}
```

Comment:
Is there anything that has changed here apart from holding onto the connections here?

Reply:
Do we actually want to hold onto these connections here?

Reply:
It is ok, now we have metrics about the connections number and we know that flask will kill the process (and thus the connections) periodically. So it is not an issue.